Code/Framework/AtomCore/AtomCore/Instance/InstanceId.h

```diff
@@ -65,7 +73,8 @@ namespace AZ
             void ToString(StringType& result) const;
 
             Uuid m_guid = Uuid::CreateNull();
-            uint32_t  m_subId = 0;
+            uint32_t m_subId = 0;
+            void* m_data = nullptr;
```

Comment:
I think m_data deserves a bit of explanation. I also think we could explicitly make this AssetData; I'm not sure how it could be used otherwise, and would be more clear.
```suggestion
            // If AssetData is included as part of the ID, that will prevent the following situation: an asset is hot-reloaded
            // and the OnAssetReloaded callback finds an old instance that uses the old asset instead of creating a
            // new instance with the reloaded asset.
            AssetData* m_assetData = nullptr;
```

Reply:
I'll add additional comments and possibly rename the variable but I intentionally made it a void pointer to dissuade someone from attempting to access it something other than a unique, raw value.